keps/2941-DRA/README.md

```diff
@@ -0,0 +1,729 @@
+# Configuration allowing device class to map to multiple canonical names
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: DynamicResourceAllocationConfig
+metadata:
+  name: "cluster-device-mapping"
+spec:
+  dynamicResources:
+  - name: whole-gpus
+    deviceClassNames:
+    - gpus.example.com          # Appears here
+  - name: fast-gpus
+    deviceClassNames:
+    - gpus.example.com          # And also here - creates ambiguity
+---
+# ClusterQueue using canonical names from the DynamicResourceAllocationConfig
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ClusterQueue
+metadata:
+  name: multi-zone-queue
+spec:
+  resourceGroups:
+  - coveredResources: ["cpu", "memory", "whole-gpus", "fast-gpus"]
+    flavors:
+    - name: zone-a-flavor
+      resources:
+      - name: whole-gpus
+        nominalQuota: 2
+    - name: zone-b-flavor
+      resources:
+      - name: fast-gpus
+        nominalQuota: 3
+```
+
+When a device class like `gpus.example.com` maps to multiple canonical names (`whole-gpus` and `fast-gpus`), Kueue must
+determine which quota to use for admission.
+
+Kueue resolves this ambiguity using a top-down flavor matching process:
+
+1. Examine flavors in the order they are listed in the ClusterQueue spec
+2. For each flavor, check if any of its resources match the canonical names that correspond to the requested device class
+3. Select the first flavor that contains a resource mapping to the device class and has sufficient quota available
+
+For example, when a workload requests `gpus.example.com`:
+- First check `zone-a-flavor` for available `whole-gpus` quota
+- If `zone-a-flavor` is exhausted, check `zone-b-flavor` for available `fast-gpus` quota  
+- If no flavors have available quota, the workload waits in queue
+
+This approach ensures predictable behavior where the ClusterQueue flavor ordering determines quota evaluation priority.
+
+### Workloads
+
+When a user submits a workload and DynamicResourceAllocation feature gate is on, Kueue will do the following:
+
+1. Claims will be read from podSpec.resourceClaims in the PodTemplateSpec.
+2. ResourceClaimSpec will be looked up either by using:
+   1. the name of the ResourceClaimTemplate or
+   2. the name of the ResourceClaim
+
+   Both ResourceClaimTemplate or ResourceClaim will be in the same namespace as the workload.
+3. From the ResourceClaimSpec, the deviceClassName will be read.
+4. Every claim, deviceClassName for each request will be looked at
+   1. For the workload a deviceClassMap will be created by:
+      1. Retrieving the singleton DynamicResourceAllocationConfig resource (named "default")
+      2. Creating a map from DeviceClass names to canonical resource names using the
+          DynamicResourceAllocationConfig.spec.dynamicResources entries
+      3. For resource claims that are already allocated, Kueue will maintain a mapping called `admittedResourceClaims`
+         where the key follows the format `{clusterQueueName}/{resourceClaimNamespace}/{resourceClaimName}`
```

Comment:
nit: this format of the key sounds like a technical detail. Actually in the cache.go we already have clustequeue caches so we could just keep the key per CQ cache. So, I consider it a technical detail which may change as we progress with the implementation.